charts/istio/istio-ingress/templates/envoy-filter.yaml

```diff
@@ -228,3 +228,44 @@ spec:
             accept_http_10: true
           upgrade_configs:
           - upgrade_type: CONNECT
+
+---
+
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: reversed-vpn
+  namespace: {{ .Release.Namespace }}
+spec:
+  configPatches:
+  - applyTo: NETWORK_FILTER
+    match:
+      context: GATEWAY
+      listener:
+        filterChain:
+          filter:
+            name: envoy.filters.network.http_connection_manager
+        name: 0.0.0.0_8132
+        portNumber: 8132
+    patch:
+      operation: MERGE
+      value:
+        name: envoy.filters.network.http_connection_manager
+        typed_config:
+          '@type': type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          route_config:
+            virtual_hosts:
+            - domains:
+              - api.*
+              name: reversed-vpn
+              routes:
+              - match:
+                  connect_matcher: {}
+                route:
+                  cluster_header: Reversed-VPN
```

Comment:
I'm not sure that we can realistically merge this change.

At the moment we configure istio in a such way that **ALL** services in the cluster (anything behind K8S `Service`) is not advertised to envoy. This means that if you need to advertise your `Service` to envoy, you have to manually set `networking.istio.io/exportTo=*` annotation.

However, end-users might deploy their own istio and by using the default installation - advertise *ALL* `Services` in the cluster. This opens a really bad security hole.

Even worse - envoy has `sds-grpc` meaning that a well-crafted request can access secrets available only to envoy by making a GRPC call after HTTP connect.

Reply:
@mvladev How does the destination rule play into this? After removing the destination rule the vpn-seed-server service did not receive traffic from istio anymore. Therefore, it seems to me that just setting the annotation is not enough to expose a service. Would that not also mean that in a user managed istio installation corresponding destination rules would still be required for the services to be exposed? I would argue that this reduces the security hole to the point that someone needs to do intentional misconfigurations.
Anyway, we could also add network policies as the targets from istio should be well known, e.g. kube-apiserver and vpn-seed-server.

What do you think?